[18.0][MIG] maintenance_plan: Another Migration to 18.0

ypapouin opened the pull request:
Continuation from https://github.com/OCA/maintenance/pull/514, I'm reusing my comment from https://github.com/OCA/maintenance/pull/514#issuecomment-3398078981

You can read that `maintenance_plan` is deprecated but merged in 17.0 https://github.com/OCA/maintenance/pull/414

Currently some other modules "depends" on it (from 14.0 to 17.0):
- `maintenance_inspection`: https://github.com/OCA/maintenance/tree/14.0/maintenance_inspection
- `maintenance_location`: https://github.com/OCA/maintenance/tree/17.0/maintenance_location
- `maintenance_plan_activity`: https://github.com/OCA/maintenance/tree/16.0/maintenance_plan_activity
- `maintenance_project_plan`: https://github.com/OCA/maintenance/tree/14.0/maintenance_project_plan

For some of them (`maintenance_inspection`,  `maintenance_location`), it would only require to drop references to `maintenance_plan` items to keep them in 18.0.

But that's not the point: the "recurrent" implementation from Odoo is javascript-only ... and its behavior has nothing to do with `maintenance_plan`. I think there has been a misunderstanding on this because the “advanced” features of the `maintenance_plan` module are much closer and more realistic to the way a maintenance workshop operates and this module should be reintroduced, that's the point of this PR.

Following are changes:

- To avoid confusion on the "recurrent" attribute, built-in fields are hidden when `maintenance_plan` is installed
- `_compute_recurring_maintenance` is always `False`
- The calendar view javascript attribute `js_class="calendar_with_recurrence"` is removed.
- The `note` field from `maintenance.request` has been dropped (use instructions fields instead)
- The `note` field from `maintenance.plan` has been replaced with the same instructions fields existing in `maintenance_request` since 17.0.

Remaining things to do:

- Migration script to convert `note` field content but how ?
- Create an intermediate wizard for the "Create request from maintenance plan" button to allow selecting a default start hour and selecting valid days of week.
- Backport this PR when merged.

grindtildeath committed:
[ADD] maintenance_plan module (#2)

pedrobaeza committed:
[FIX] maintenance_plan: README try me on runbot link

bestmazzo committed:
[MIG] maintenance_plan: Migration to 11.0

dalonsod committed:
[FIX] maintenance_plan: prevent preventive request cron crashing when equipment maintenance team is not filled

[ADD] icon.png

dalonsod committed:
[MIG] maintenance_plan: Migration to 12.0

[UPD] Update maintenance_plan.pot

AdriaGForgeFlow committed:
[12.0][IMP] maintenance_plan

[ADD] instructions and notes to Plan

- Add menus for maintenance plans.
- Set active button on maintenance plans.

improve navigation to maintenance plans.

[ADD] post-migration script

lromero-solvos committed:
Added translation using Weblate (Spanish)

lromero-solvos committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (30 of 30 strings)

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/es/

[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

dalonsod committed:
[FIX] maintenance_plan: cron generated request should inherit equipment company

maintenance_plan 12.0.1.1.0

dalonsod committed:
[FIX] maintenance_plan: next_maintenance_date should be stored

Commit:
[12.0][FIX] default equipment_id in plans

etobella committed:
[11.0][IMP] maintenance_plan: Return the created requests

etobella committed:
[12.0][FIX] maintenance_project_plan

etobella committed:
[12.0][IMP] maintenance_plan: Add team on plans

[UPD] Update maintenance_plan.pot

[UPD] README.rst

maintenance_plan 12.0.3.0.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/

dalonsod committed:
[FIX] maintenance_plan: prevent False as display name when name is empty

[UPD] Update maintenance_plan.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/

dalonsod committed:
Translated using Weblate (Spanish)

Currently translated at 98.4% (63 of 64 strings)

Translation: maintenance-12.0/maintenance-12.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-12-0/maintenance-12-0-maintenance_plan/es/

NuriaXifre committed:
[IMP] maintenance_plan: black, isort

NuriaXifre committed:
[13.0][MIG] maintenance_plan

[UPD] Update maintenance_plan.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

Commit:
[12.0][IMP] maintenance_plan: no equipment

[UPD] Update maintenance_plan.pot

maintenance_plan 13.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

pre-commit update

AaronHForgeFlow committed:
[FIX]maintenance_plan migration scripts not applicable in v13

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

LoisRForgeFlow committed:
[13.0][IMP] maintenance_plan: make maintenance plan multi-company compatible.
includes migration scripts

LoisRForgeFlow committed:
maintenance_plan: manual version bump

[UPD] Update maintenance_plan.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-13.0/maintenance-13.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_plan/

dalonsod committed:
[IMP] maintenance_plan: black, isort, prettier

dalonsod committed:
[MIG] maintenance_plan: Migration to 14.0

[UPD] Update maintenance_plan.pot

[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

etobella committed:
[FIX] maintenance_plan: Remake the computation of next date
* Use latest request_date both for done and todo searches.
* When there is a latest done apply interval

Co-authored-by: Lois Rilo <lois.rilo@forgeflow.com>
Co-authored-by: LoisRForgeFlow <LoisRForgeFlow@users.noreply.github.com>

maintenance_plan 14.0.1.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

LoisRForgeFlow committed:
[IMP] maintenance_plan: next maintenance date tweaks

* only consider request after the start maintenance date.
* todo request ordered asc.
* add more tests.

maintenance_plan 14.0.1.0.2

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (69 of 69 strings)

Translation: maintenance-14.0/maintenance-14.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-14-0/maintenance-14-0-maintenance_plan/it/

BernatPForgeFlow committed:
[IMP] maintenance_plan: black, isort, prettier

BernatPForgeFlow committed:
[MIG] maintenance_plan: Migration to 15.0

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[UPD] README.rst

mariadforgeflow committed:
[FIX] maintenance_plan: fix new request date when plan start date is later

OCA-git-bot committed:
maintenance_plan 15.0.1.0.1

MiquelRForgeFlow committed:
[FIX] maintenance_plan: avoid error if planning_step is not defined

Also, clean-up of useless from_string() calls.

mariadforgeflow committed:
[IMP] maintenance_plan: add chatter

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.1.1

sergiocorato committed:
[15.0][FIX] maintenance_plan generate requests for inactive equipment

OCA-git-bot committed:
maintenance_plan 15.0.1.1.2

mymage committed:
Translated using Weblate (Italian)

Currently translated at 67.9% (72 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (106 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
[REF] moving function get_relativedelta to method

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 58.4% (62 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/pt/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (106 of 106 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/pt/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Set planned_hours in request from plan if maintenance_timesheet is installed.

kluna1998 committed:
[FIX] maintenance_plan: fix maintence request form view

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (111 of 111 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Add Generate requests for current threshold button from plan view form.

Add button from maintenance plan to allow generate maintenance requests
without waiting for cron.

TT41729

victoralmau committed:
[IMP] maintenance_plan: Add instructions to request report.

TT41671

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.3.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.4.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.5.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.5.1

victoralmau committed:
[IMP] maintenance_plan: Change tests to allow inheritance

TT41735

victoralmau committed:
[IMP] maintenance_plan: Skip assigned mail + Activity mail of the requests created from plans

TT41870

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
maintenance_plan 15.0.1.6.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

OCA-git-bot committed:
maintenance_plan 15.0.1.6.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

victoralmau committed:
[IMP] maintenance_plan: Add rules to maintenance.plan to view records of the equipments they follow

TT42535

OCA-git-bot committed:
maintenance_plan 15.0.1.7.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 85.9% (98 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[IMP] maintenance_plan: Allow to generate generic plans for multiple equipments defined with a domain

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (114 of 114 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/es/

etobella committed:
[FIX] maintenance_plan: The search must work with newIds

etobella committed:
[IMP] maintenance_plan: Add Team filter

Commit:
[UPD] Update maintenance_plan.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

victoralmau committed:
[IMP] maintenance_plan: Add freeze_time to prevent errors in tests

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[FIX] maintenance_plan: adapt tests

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (118 of 118 strings)

Translation: maintenance-15.0/maintenance-15.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-15-0/maintenance-15-0-maintenance_plan/it/

alexeirivera87 committed:
[MIG] maintenance_plan: Migration to 16.0

Commit:
[UPD] Update maintenance_plan.pot

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (118 of 118 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_plan/es/

FernandoRomera committed:
[IMP] maintenance_plan: pre-commit auto fixes

FernandoRomera committed:
[MIG] maintenance_plan: Migration to 17.0

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update maintenance_plan.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-17.0/maintenance-17.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_plan/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (115 of 115 strings)

Translation: maintenance-17.0/maintenance-17.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_plan/it/

vincent-hatakeyama committed:
[FIX] maintenance_plan: fix the use of id in act_window

Without this fix, the smart button using the act_window fails with an
error.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (115 of 115 strings)

Translation: maintenance-17.0/maintenance-17.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_plan/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (115 of 115 strings)

Translation: maintenance-17.0/maintenance-17.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_plan/it/

carlos-lopez-tecnativa committed:
[IMP] base_maintenance: Fine-tuning of the form view
after this commit https://github.com/odoo/odoo/commit/3bd841086d92246e55771785bcb69501af712ab6 the maintenance request form view now has an independent page. This commit adapts the view to align with the new page structure.

Commit:
[UPD] Update maintenance_plan.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: maintenance-17.0/maintenance-17.0-maintenance_plan
Translate-URL: https://translation.odoo-community.org/projects/maintenance-17-0/maintenance-17-0-maintenance_plan/

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] maintenance_plan: Define module as deprecated in readme

https://github.com/OCA/maintenance/pull/414#issuecomment-2668929617

OCA-git-bot committed:
[BOT] post-merge updates

ypapouin committed:
[IMP] maintenance_plan: Remove "deprecated" status

ypapouin committed:
[IMP] maintenance_plan: Path added to actions

ypapouin committed:
[FIX] maintenance_plan: Odoo dependency flaw

ypapouin committed:
[IMP] maintenance_plan: pre-commit auto fixes

ypapouin committed:
[MIG] maintenance_plan: Migration to 18.0

ThiagoMForgeFlow commented:
I’m not sure if the best way to maintain the functionalities of the child modules is to migrate this module and make Odoo’s standard functionality invisible. Perhaps we should modify the child modules so that instead of inheriting from maintenance_plan, they inherit from base_maintenance, and then refactor their functionalities. What do you think @pedrobaeza ?

etobella commented:
Well, after some time struggling with the Odoo way and the maintenance plan way I think I can share my POV.

1- Odoo way is simpler, however, it works well only in controlled environments. For example, If I need to make a maintenance plan for reviews every 3 years, Odoo allows me to do it, but it creates some preventive tasks in advance that have no sense IMO.
2- Odoo way requires the creation of the first task. at some point, we added the option of autodetection of equipments. That was much better IMO.
3- Maintenance plan is complicated might be harder to configure
4- Maintenance plan allows me to create tasks in advance at the exact point I want to.

IMO, we should keep both options in the system and make maintenance_plan work in what makes sense, complex environments. It means that we shouldn't hide the odoo way, however, I would recommend to use maintenance plan if possible. About the migrations, we should need to do some stuff to avoid the execution of the migration scripts, but I have an idea of how to do it.

pedrobaeza commented:
You can explain that on the CONTEXT.md file of this module.

ThiagoMForgeFlow commented on maintenance_plan/models/maintenance_plan.py:
```diff
@@ -0,0 +1,272 @@
+            "time": safe_eval.time,
+        }
+
+    def name_get(self):
```
In v18 this method no longer exists; you should inherit the _compute_display_name method instead.

ThiagoMForgeFlow commented on maintenance_plan/models/maintenance_plan.py:
```diff
@@ -0,0 +1,272 @@
+        help="Paste the url of your Google Slide. "
+        "Make sure the access to the document is public.",
+    )
+    instruction_text = fields.Html("Text")
```
To migrate the data that was stored in the note field up to version 17, we need a migration script in the folder migrations/18.0.1.0/pre-migration.py with the following code:
```

from openupgradelib import openupgrade

@openupgrade.migrate()
def migrate(env, version):
    if openupgrade.column_exists(env.cr, "maintenance_plan", "note"):
        openupgrade.rename_fields(
            env,
            [
                (
                    "maintenance.plan",
                    "maintenance_plan",
                    "note",
                    "instruction_text",
                ),
            ],
        )
```

ThiagoMForgeFlow commented on maintenance_plan/views/maintenance_equipment_views.xml:
```diff
@@ -0,0 +1,71 @@
+                </field>
+            </xpath>
+            <field name="maintenance_team_id" position="after">
+                <field name="maintenance_team_required" invisible="True" />
```
In v18 it is no longer necessary to add fields with the invisible="true" attribute, you can remove this from the view.

ThiagoMForgeFlow commented:
> Well, after some time struggling with the Odoo way and the maintenance plan way I think I can share my POV.
> 
> 1- Odoo way is simpler, however, it works well only in controlled environments. For example, If I need to make a maintenance plan for reviews every 3 years, Odoo allows me to do it, but it creates some preventive tasks in advance that have no sense IMO. 2- Odoo way requires the creation of the first task. at some point, we added the option of autodetection of equipments. That was much better IMO. 3- Maintenance plan is complicated might be harder to configure 4- Maintenance plan allows me to create tasks in advance at the exact point I want to.
> 
> IMO, we should keep both options in the system and make maintenance_plan work in what makes sense, complex environments. It means that we shouldn't hide the odoo way, however, I would recommend to use maintenance plan if possible. About the migrations, we should need to do some stuff to avoid the execution of the migration scripts, but I have an idea of how to do it.

Hi @etobella, have you been able to spend some time on the proposal for making this module’s functionalities coexist with the ones provided by Odoo?

etobella commented:
@ThiagoMForgeFlow I didn't, there were no more answer on the topic, however, it might have sense IMO to restore it

ThiagoMForgeFlow commented:
I agree that the module should be kept in version 18 @etobella. But I don’t think Odoo’s standard functionality should be hidden; perhaps you could add a setting to decide whether it is visible or not at the equipment or company level. What do you think @ypapouin ?

etobella commented:
I do agree with your approach

ypapouin commented:
I agree that the fields could easily be hidden using an option (company/equipment), but the biggest problem comes from the calendar. It uses a specific JavaScript implementation, and in fact, Odoo's entire recurring implementation is done in this calendar.

Honestly, I don't know how this option could be implemented to disable "maintenance_plan" inherited views for a specific company/equipment. (I don't really see the point of having an option to hide or show this feature, since once you install the “maintenance_plan” module, you accept that Odoo's built-in recurrent features will not be visible.)

etobella commented:
I think both approaches have their own target, and might even have sense to have them together in some specific environments.

I think we shouldn't remove odoo maintenance behavior by default. If that is a problem, we can have a second module for that.

ypapouin commented:
> we can have a second module for that.

I'm Ok with that, I will extract every "hide/disable" view items to a specific module and add a `res.config.settings` to install it.
Which name for a such module ? `maintenance_plan_only` ?

etobella commented:
Yes, it seems correct to me.

ypapouin committed:
[ADD] maintenance_plan_only

ypapouin committed:
[IMP] maintenance_plan: Replace note with instructions

- `maintenance.request` views are now set in their own file
- `note` field replaced with existing combo `instruction_type,instruction_pdf,instruction_google_slide,instruction_text`

ypapouin committed:
[IMP] maintenance_plan: Setting added to install "maintenance_plan_only" module

ypapouin commented:
As requested, everything related to hide Odoo builtin recurrent maintenance has been moved to a new module.
The last commit adds the `maintenance_plan_only` module that can be enabled using a setting:
<img width="983" height="550" alt="image" src="https://github.com/user-attachments/assets/dcdafbfe-95a2-4487-822e-9425b6f20116" />

cvinh commented on maintenance_plan/views/maintenance_equipment_views.xml:
```diff
@@ -0,0 +1,68 @@
+                </button>
+            </div>
+            <xpath expr="//group[@name='statistics']" position="attributes">
+                <attribute name="invisible">True</attribute>
```
Does anybody remember why the statistic data have been hidden ?

ThiagoMForgeFlow replied:
Hi @cvinh, after checking the commit history I can see that it was originally added in version 12, replacing the group maintenance. Then, in the migrations for versions 13 and 17, the xpath was adjusted to match each Odoo version. However, looking at Odoo version 18, I’m not sure it still makes sense to hide these fields.

original implemenatation: [12_imp](https://github.com/OCA/maintenance/commit/10cd585153bf9bda8807f93e1332982157124ae9#diff-c5ed3c8212de850d39e065143984c883ee5e41983ae4ac2ff9a20d17c18c3df2R18)

Migration commit 13: [13_mig](https://github.com/OCA/maintenance/commit/6a5739d3a822abc734b323db29ce4690fedeace8#diff-c5ed3c8212de850d39e065143984c883ee5e41983ae4ac2ff9a20d17c18c3df2R18)

Migration commit 17: [17_mig](https://github.com/OCA/maintenance/commit/9f99863b12f257cc9ab2f4f34aa3f7cafd4eb0d5#diff-c5ed3c8212de850d39e065143984c883ee5e41983ae4ac2ff9a20d17c18c3df2R23)

v18 fields: 
<img width="864" height="348" alt="2025-11-24_09-44" src="https://github.com/user-attachments/assets/eb1b09fd-e3ce-4f46-ae4c-499982e14b08" />

ThiagoMForgeFlow commented:
Hi @etobella and @pedrobaeza What is your final opinion?

pedrobaeza commented:
I let @etobella to give his opinion, as we are going to the native feature.

etobella reviewed (approved):
Let's proceed with it

/ocabot merge nobump
